packages/typespec-client-generator-core/src/package.ts

```diff
@@ -606,9 +607,11 @@ function getSdkMethods<TServiceOperation extends SdkServiceOperation>(
   const diagnostics = createDiagnosticCollector();
   const retval: SdkMethod<TServiceOperation>[] = [];
   for (const operation of listOperationsInOperationGroup(context, client)) {
```

Comment:
i think you should change `listOperationsInOperationGroup`, instead of filter operation in `pacakge.ts` and `types.ts`.